_posts/2024-08-22-boost-vector-search-with-css.md

```diff
@@ -0,0 +1,574 @@
+    <tr>
+      <td rowspan="2">Lucene</td>
+      <td rowspan="2">381</td>
+      <td>1</td>
+      <td>14</td>
+      <td>16</td>
+      <td>18</td>
+      <td>43</td>
+      <td>45.37</td>
+      <td>0.97</td>
+    </tr>
+    <tr>
+      <td>4</td>
+      <td>28</td>
+      <td>34</td>
+      <td>43</td>
+      <td>87</td>
+      <td>48.79</td>
+      <td>0.97</td>
+    </tr>
+    <tr>
+      <td rowspan="2">NMSLIB</td>
+      <td rowspan="2">383</td>
+      <td>1</td>
+      <td>10</td>
+      <td>12</td>
+      <td>14</td>
+      <td>41</td>
+      <td>45.21</td>
+      <td>0.97</td>
+    </tr>
+    <tr>
+      <td>4</td>
+      <td>23</td>
+      <td>25</td>
+      <td>29</td>
+      <td>75</td>
+      <td>45.87</td>
+      <td>0.97</td>
+    </tr>
+    <tr>
+      <td rowspan="2">Faiss</td>
+      <td rowspan="2">381</td>
+      <td>1</td>
+      <td>15</td>
+      <td>16</td>
+      <td>17</td>
+      <td>44</td>
+      <td>48.68</td>
+      <td>0.97</td>
+    </tr>
+    <tr>
+      <td>4</td>
+      <td>23</td>
+      <td>26</td>
+      <td>32</td>
+      <td>79</td>
+      <td>47.19</td>
+      <td>0.97</td>
+    </tr>
+  </tbody>
+</table>
+
+### Results comparison 
+
+For simplicity, we'll focus on the p90 metric from our experiments with a single search client, because this metric captures the performance of long-running vector search queries.
+
+#### Service time comparison (p90)
+
+|k-NN engine	|Concurrent segment search disabled	|Concurrent segment search enabled (Lucene default number of slices)	|% Improvement	|Concurrent segment search with max slice count = 2	|% Improvement	|Concurrent segment search with max slice count =4	|% Improvement	|Concurrent segment search with max slice count = 8	|% Improvement	|
+|---	|---	|---	|---	|---	|---	|---	|---	|---	|---	|
+|Lucene	|37	|15	|59.5	|16	|56.8	|15.9	|57	|16	|56.8	|
+|NMSLIB	|35	|14	|60	|23	|34.3	|15	|57.1	|12	|65.7	|
+|Faiss	|37	|14	|62.2	|22	|40.5	|15	|59.5	|16	|56.8	|
+
+
+#### CPU utilization comparison
+
+|k-NN engine	|Concurrent segment search disabled	|Concurrent segment search enabled (Lucene default number of slices)	|% Additional CPU utilization	|Concurrent segment search with max slice count = 2	|% Additional CPU utilization	|Concurrent segment search with max slice count =4	|% Additional CPU utilization	|Concurrent segment search with max slice count = 8	|% Additional CPU utilization	|
+|---	|---	|---	|---	|---	|---	|---	|---	|---	|---	|
+|Lucene	|11	|47	|36	|41	|30	|49	|38	|43	|32	|
+|NMSLIB	|10	|38	|28	|16	|6	|29	|19	|41	|31	|
+|Faiss	|10	|34	|24	|19	|9	|30	|20	|44	|34	|
+
+
+As demonstrated by our performance benchmarks, enabling concurrent segment search with the default slice count delivers at least a **60% improvement** in vector search service time while requiring only **25-35% more CPU**. This increase in CPU utilization is expected because concurrent segment search runs on more CPU threads---the number of threads is equal to twice the number of CPU cores.
```

Comment:
```suggestion
As demonstrated by our performance benchmarks, enabling concurrent segment search with the default slice count delivers at least a **60% improvement** in vector search service time while requiring only **25--35% more CPU**. This increase in CPU utilization is expected because concurrent segment search runs on more CPU threads---the number of threads is equal to twice the number of CPU cores.
```